pkg/azuredisk/azure.go

```diff
@@ -19,27 +19,73 @@ package azuredisk
 import (
 	"fmt"
 	"os"
+	"reflect"
 
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/klog"
 	"k8s.io/legacy-cloud-providers/azure"
 )
 
+var (
+	DefaultAzureCredentialFileEnv = "AZURE_CREDENTIAL_FILE"
+	DefaultCredFilePath           = "/etc/kubernetes/azure.json"
+)
+
 // GetCloudProvider get Azure Cloud Provider
-func GetCloudProvider() (*azure.Cloud, error) {
-	credFile, ok := os.LookupEnv("AZURE_CREDENTIAL_FILE")
-	if ok {
-		klog.V(2).Infof("AZURE_CREDENTIAL_FILE env var set as %v", credFile)
-	} else {
-		credFile = "/etc/kubernetes/azure.json"
-		klog.V(2).Infof("use default AZURE_CREDENTIAL_FILE env var: %v", credFile)
+func GetCloudProvider(kubeconfig string) (*azure.Cloud, error) {
+	kubeClient, err := getKubeClient(kubeconfig)
+	if err != nil && !os.IsNotExist(err) && err != rest.ErrNotInCluster {
+		return nil, fmt.Errorf("failed to get KubeClient: %v", err)
+	}
+
+	az := &azure.Cloud{}
+	if kubeClient != nil {
+		klog.V(2).Infof("reading cloud config from secret")
+		az.KubeClient = kubeClient
+		az.InitializeCloudFromSecret()
+	}
+
+	if reflect.DeepEqual(az, &azure.Cloud{}) {
```

Comment:
In order to follow the thought that we do not verify the azure.Cloud in func`GetCloudProvider()`, changed the judgment condition.  

When the user provides a wrong kubeconfig, kubeClient is still created. az will be initialize. Although it is invalid, it is created successfully, then return it.